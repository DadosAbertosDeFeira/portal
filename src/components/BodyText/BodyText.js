/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';

const classNames = (size, bold) => {
  const classes = [];

  if (size === 'small') classes.push('text-xs tablet:text-base');
  else if (size === 'medium') classes.push('text-sm tablet:text-base');
  if (bold) classes.push('font-bold');

  return classes.join(' ');
};

const BodyText = ({ className, children, size, bold, testId, ...props }) => (
  <p
    data-testid={testId}
    className={`text-black ${classNames(size, bold)} ${className}`}
    {...props}
  >
    {children}
  </p>
);

BodyText.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['medium', 'small']),
  bold: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.array,
    PropTypes.object,
  ]),
  testId: PropTypes.string,
};

BodyText.defaultProps = {
  className: '',
  size: 'medium',
  bold: false,
  children: '',
  testId: '',
};

export default BodyText;
