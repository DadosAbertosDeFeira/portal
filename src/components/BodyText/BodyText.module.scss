@import '../../styles/variables.scss';

.bodyText {
  color: $color-black;
  font-weight: 400;

  &.bold {
    font-weight: 700;
  }
}

.medium {
  font-size: 0.87rem;
  line-height: 1.31rem;
}

.small {
  font-size: 0.69rem;
  line-height: 1.03rem;
}
