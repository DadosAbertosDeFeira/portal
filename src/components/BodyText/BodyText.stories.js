/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import BodyText from './BodyText';

export const BodyTextTemplate = (args) => <BodyText {...args} />;
BodyTextTemplate.args = {
  children: 'Portal Dados Abertos',
  size: 'medium',
};

export const Small = BodyTextTemplate.bind({});
Small.args = {
  ...BodyTextTemplate.args,
  size: 'small',
};

export const Medium = BodyTextTemplate.bind({});
Medium.args = {
  ...BodyTextTemplate.args,
  size: 'medium',
};

export default {
  title: 'Components/BodyText',
  component: BodyText,
};
