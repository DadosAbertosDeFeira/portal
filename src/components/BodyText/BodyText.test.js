/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render } from '@testing-library/react';

import BodyText from './BodyText';

describe('<BodyText />', () => {
  let sharedProps;

  beforeEach(() => {
    sharedProps = {
      testId: 'paragraph',
    };
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <BodyText {...sharedProps}>Testing</BodyText>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly default size and bold', () => {
    const { asFragment } = render(
      <BodyText {...sharedProps} bold>
        Testing
      </BodyText>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly small', () => {
    const { asFragment } = render(
      <BodyText {...sharedProps} size="small">
        Testing
      </BodyText>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly small and bold', () => {
    const { asFragment } = render(
      <BodyText {...sharedProps} bold size="small">
        Testing
      </BodyText>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with html alternative attributes', () => {
    const { asFragment } = render(
      <BodyText {...sharedProps} tabIndex={0}>
        Testing
      </BodyText>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
