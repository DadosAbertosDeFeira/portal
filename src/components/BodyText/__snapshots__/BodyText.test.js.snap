// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyText /> renders correctly 1`] = `
<DocumentFragment>
  <p
    class="text-black text-sm tablet:text-base "
    data-testid="paragraph"
  >
    Testing
  </p>
</DocumentFragment>
`;

exports[`<BodyText /> renders correctly default size and bold 1`] = `
<DocumentFragment>
  <p
    class="text-black text-sm tablet:text-base font-bold "
    data-testid="paragraph"
  >
    Testing
  </p>
</DocumentFragment>
`;

exports[`<BodyText /> renders correctly small 1`] = `
<DocumentFragment>
  <p
    class="text-black text-xs tablet:text-base "
    data-testid="paragraph"
  >
    Testing
  </p>
</DocumentFragment>
`;

exports[`<BodyText /> renders correctly small and bold 1`] = `
<DocumentFragment>
  <p
    class="text-black text-xs tablet:text-base font-bold "
    data-testid="paragraph"
  >
    Testing
  </p>
</DocumentFragment>
`;

exports[`<BodyText /> renders correctly with html alternative attributes 1`] = `
<DocumentFragment>
  <p
    class="text-black text-sm tablet:text-base "
    data-testid="paragraph"
    tabindex="0"
  >
    Testing
  </p>
</DocumentFragment>
`;
